### PR TITLE
Deprecate certain cftime frequency strings following pandas

### LINKED ISSUE #8394: Update cftime frequency strings in line with recent updates in pandas
### What is your issue?

Pandas has introduced some deprecations in how frequency strings are specified:

- Deprecating `"A"`, `"A-JAN"`, etc. in favor of `"Y"`, `"Y-JAN"`, etc. (https://github.com/pandas-dev/pandas/pull/55252)
- Deprecating `"AS"`, `"AS-JAN"`, etc. in favor of `"YS"`, `"YS-JAN"`, etc. (https://github.com/pandas-dev/pandas/pull/55479)
- Deprecating `"Q"`, `"Q-JAN"`, etc. in favor of `"QE"`, `"QE-JAN"`, etc.  (https://github.com/pandas-dev/pandas/pull/55553)
- Deprecating `"M"` in favor of `"ME"` (https://github.com/pandas-dev/pandas/pull/54061)
- Deprecating `"H"` in favor of `"h"` (https://github.com/pandas-dev/pandas/pull/54939)
- Deprecating `"T"`, `"S"`, `"L"`, and `"U"` in favor of `"min"`, `"s"`, `"ms"`, and `"us"` (https://github.com/pandas-dev/pandas/pull/54061).

It would be good to carry these deprecations out for cftime frequency specifications to remain consistent.

### PR DESCRIPTION
**Pull request by spencerkclark**
Following several upstream PRs in pandas, this PR deprecates cftime frequency strings `"A"`, `"AS"`, `"Q"`, `"M"`, `"H"`, `"T"`, `"S"`, `"L"`, and `"U"` in favor of `"Y"`, `"YS"`, `"QE"`, `"ME"`, `"h"`, `"min"`, `"s"`, `"ms"`, and `"us"`.  Similarly following pandas, it makes a breaking change to have `infer_freq` return the latter frequencies instead of the former.

There are a few places in the tests and one place in the code where we need some version-specific logic to retain support for older pandas versions.  @aulemahal it would be great if you could take a look to make sure that I handled this breaking change properly / fully in the `date_range_like` case.

I also took the liberty to transition to using `"Y"`, `"YS"`, `"h"`, `"min"`, `"s"`, `"ms"`, `"us"`, and `"ns"` within our code, tests, and documentation to reduce the amount of warnings emitted.  I have held off on switching to `"QE"`, `"ME"`, and anchored offsets involving `"Y"` or `"YS"` in pandas-related code since those are not supported in older versions of pandas.

The deprecation warning looks like this:
```
>>> xr.cftime_range("2000", periods=5, freq="M")
<stdin>:1: FutureWarning: 'M' is deprecated and will be removed in a future version. Please use 'ME' instead of 'M'.
CFTimeIndex([2000-01-31 00:00:00, 2000-02-29 00:00:00, 2000-03-31 00:00:00,
             2000-04-30 00:00:00, 2000-05-31 00:00:00],
            dtype='object', length=5, calendar='standard', freq='ME')
```

- [x] Closes #8394
- [x] Addresses the `convert_calendar` and `date_range_like` test failures in #8091
- [x] Tests added
- [x] User visible changes (including notable bug fixes) are documented in `whats-new.rst`

### CODE REVIEW

File: xarray/coding/cftime_offsets.py [hunk 1]
```diff
@@ -965,84 +979,84 @@ def cftime_range(
...
-    | AS, YS | Year-start frequency     |
+    | YS     | Year-start frequency     |
     +--------+--------------------------+
-    | Q      | Quarter-end frequency    |
+    | QE     | Quarter-end frequency    |
     +--------+--------------------------+
     | QS     | Quarter-start frequency  |
     +--------+--------------------------+
-    | M      | Month-end frequency      |
+    | MS     | Month-end frequency      |
```
Comment (aulemahal) [inline xarray/coding/cftime_offsets.py:990]:
```suggestion
    | ME     | Month-end frequency      |
```

File: xarray/tests/test_cftime_offsets.py [hunk 2]
```diff
@@ -1363,20 +1380,53 @@ def test_date_range_errors() -> None:
...
+        expected_pandas_freq = freq
+
     source = date_range(start, periods=12, freq=freq, calendar=cal_src)
 
     out = date_range_like(source, cal_tgt, use_cftime=use_cftime)
 
     assert len(out) == 12
-    assert infer_freq(out) == freq
+
+    print(out)
```
Comment (spencerkclark) [inline xarray/tests/test_cftime_offsets.py:1425]:
```suggestion
```

File: xarray/coding/cftime_offsets.py (hunk 1 above)
Comment (spencerkclark) [reply xarray/coding/cftime_offsets.py:990]: Good catch, thanks!

**Comment by aulemahal** [comment]
I am on board with these changes! Thanks.

**Pull request merged**